.azure-pipelines/client.yml

```diff
@@ -58,57 +58,33 @@ jobs:
       displayName: 'Component Detection'
 
     - task: Maven@3
-      displayName: 'Generate project site'
+      displayName: 'Generate Maven project site, including JavaDocs and CheckStyle reports'
       inputs:
         mavenPomFile: $(pomFile)
-        options: '--batch-mode'
+        options: '--batch-mode -DskipTests'
         mavenOptions: '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
         javaHomeOption: 'JDKVersion'
         jdkVersionOption: '1.11'
         jdkArchitectureOption: 'x64'
         publishJUnitResults: false
-        goals: 'site'
+        goals: 'site:site site:stage'
 
     - task: Maven@3
       displayName: 'Install packages to local cache and generate SpotBugs report'
       inputs:
         mavenPomFile: $(pomFile)
-        options: '--batch-mode -Pspotbugs-reporting -DskipTests -Dgpg.skip'
+        options: '--batch-mode -Pextra-modules -DskipTests -Dgpg.skip'
         mavenOptions: '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
         javaHomeOption: 'JDKVersion'
         jdkVersionOption: '1.11'
         jdkArchitectureOption: 'x64'
         publishJUnitResults: false
```

Comment:
If possible we should try to collapse all the calls into one Maven task with just a ordered list of goals, this will help eliminate a bunch of duplication in the yaml files.

Reply:
Having them separate is easier. The first task requires extra modules, and we don't want these extra modules included in the second task (as otherwise they will be included in the reporting step). At least with this PR we reduce from four tasks down to two :-)

Reply:
For my information does having the extra modules hurt anything? My biggest thing is to try and eliminate unnecessary duplication of parameters and such, but if you tell me that there is a good reason to keep them separate that makes sense to me.

Reply:
The reason you would want separate steps is to not include things like our tooling, templates, etc in our generated reports for spotbugs, checkstyle, and javadoc. It gives an unprofessional look to include non-SDK API inside these reports (most particularly the javadoc).

Reply:
We could also update the tool config for JavaDoc, SpotBugs, and CheckStyle to ignore the relevant packages, which would mean we can always run with extra-modules, and that they should hopefully never turn up in any of our reports.

Reply:
How are they being excluded today? Given that they are building the same pom file I'd expect them to be included either way.

Reply:
They aren't being excluded today, for example, you can see the template package in the JavaDoc here: https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-java/site/apidocs/index.html

Reply:
Regardless of excluding some projects, is there still a reason all these goals cannot be part of one call the to Maven task?

Reply:
That's the point I'm making. They can be merged as I'm excluding the packages now.

Reply:
Actually I think I finally understand what you are doing. For the javadoc/site stage you are setting `extra-modules` to exclude anything that isn't a shipping product. That makes sense to me and that is a good enough reason to me to have to different Maven calls. 

Perhaps I'm missing it but where did the checkstyle goal go?

Reply:
Checkstyle is run as part of maven `site:site`. I will be updating the PR momentarily and we'll clarify from there.

Reply:
lol :) sounds good.

Reply:
OK I think I understand what you were saying now. Since you have combined these all into one step we now are not running the stylecheck and spotbug checks on the "extra-modules" (e.g. template and tools). While that might be OK do you think it is worth running the checks on those as well? I'd worry a little bit that we might end up putting something we don't want to under extra-modules. Maybe we can mitigate that a little by naming it something like "non-shipping-modules".

Reply:
I'm ok with the rename - but I'm also fine with 'extra-modules'.

Reply:
Thanks I would prefer the rename to make it clear what it is for.